Automattic/pocket-casts-android
# Address parts of the Engage SDK integration feedback

## Description

This PR addresses part of the feedback from Google:
- Broadcast Receivers are added to the manifest.
- Publish Status is updated.
- Episode progress percentage is now correctly calculated.

Context: pdeCcb-6Ck-p2#comment-5275

I'll work on the sign in cluster separately.

## Testing Instructions

1. Use `debug` or `debugProd` variant.
2. Open the app and sign in.
3. Minimize the app.
4. Go through the first 4 steps from [the integration workflow](https://developer.android.com/guide/playcore/engage/workflow).
5. Don't mind the information about using delete API. I confirmed it with Google.
6. Verify that publish status is updated.
7. I'm couldn't verify that API accepted progress percentage. It is not displayed anywhere in the UI. I checked it only by logging the values. You can apply [this patch] if you want to verify it as well.
8. Inspect the `AndroidManifest` and verify that the [required receivers](https://developer.android.com/guide/playcore/engage/listen#broadcast-intents) are registered.

## Screenshots or Screencast 


## Checklist
 
#### I have tested any UI changes...
